middleware/http_tracing.go

```diff
@@ -10,11 +11,22 @@ import (
 )
 
 // Tracer is a middleware which traces incoming requests.
-type Tracer struct{}
+type Tracer struct {
```

Comment:
I'm wondering if we should rename `Tracer`, but can't really come up with a compelling enough alternative:

- RequestTracer
- OpentracingMiddleware
- TracingRequestHandler

🤢